test/shared/src/main/scala/zio/test/mock/Mock.scala

```diff
@@ -0,0 +1,212 @@
+package zio.test.mock
+
+import zio.{ IO, Promise, Ref, ZIO }
+
+trait Mock {
+
+  def invoke[E0, A0, A](method: Method[A, A0])(value: A): IO[E0, A0]
```

Comment:
Maye we should rename these to `apply`? The overloads should work fine, then we'll have syntax like:

```scala
mock(Get)(1, 2, 3)
```

Which would be nicer than:

```scala
mock.invoke3(Get)(1, 2, 3)
```

Reply:
Really nice for users not to have to worry about the arity of the methods they are calling.